src/lib/components/event/event-summary-row.svelte

```diff
@@ -38,103 +53,209 @@
   import EventLink from './event-link.svelte';
   import MetadataDecoder from './metadata-decoder.svelte';
 
-  export let event: IterableEvent;
-  export let group: EventGroup | undefined = undefined;
-  export let initialItem: IterableEvent | undefined;
-  export let index = 0;
-  export let compact = false;
-  export let expanded = false;
-  export let typedError = false;
-  export let active = false;
-  export let onRowClick: () => void = () => {};
+  interface Props {
+    event: WorkflowEvent;
+    group?: EventGroup;
+    initialItem: WorkflowEvent;
+    index: number;
+    compact?: boolean;
+    typedError?: boolean;
+    active?: boolean;
+    expanded?: boolean;
+    onRowClick?: () => void;
+  }
 
-  $: selectedId = isEventGroup(event)
-    ? Array.from(event.events.keys()).shift()
-    : event.id;
+  const {
+    event,
+    group,
+    initialItem,
+    index,
+    compact = false,
+    typedError = false,
+    active = false,
+    expanded: expandedProp = false,
+    onRowClick = () => {},
+  }: Props = $props();
 
-  $: ({ workflow, run, namespace } = $page.params);
-  $: href = routeForEventHistoryEvent({
-    eventId: event.id,
-    namespace,
-    workflow,
-    run,
-  });
-  $: attributes = formatAttributes(event);
+  let expanded = $state(expandedProp);
+  let primaryLocalAttribute = $state<SummaryAttribute | undefined>(undefined);
 
-  $: currentEvent = isEventGroup(event) ? event.events.get(selectedId) : event;
-  $: elapsedTime = formatDistanceAbbreviated({
-    start: initialItem?.eventTime,
-    end: isEventGroup(event)
-      ? event.initialEvent?.eventTime
-      : currentEvent.eventTime,
-    includeMillisecondsForUnderSecond: true,
-  });
+  type DecodedLocalActivity = {
+    details?: {
+      data?: {
+        payloads?: Payload[];
+      };
+    };
+  };
 
-  $: duration = isEventGroup(event)
-    ? formatDistanceAbbreviated({
-        start: event.initialEvent?.eventTime,
-        end: event.lastEvent?.eventTime,
-        includeMillisecondsForUnderSecond: true,
-      })
-    : '';
+  export const decodeLocalActivity = async (
```

Comment:
Let's not make this an export if it's in this component. Exports in .svelte files only work in <script module> tags